File: backend/app/schemas/notifications.py
"""站内消息中心 Schemas（Module 7B / D-074）。"""
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class NotificationOut(BaseModel):
    id: uuid.UUID
    type: str
    channel: str
    title: str
    content: str
    is_read: bool
    read_at: datetime | None
    created_at: datetime
    expires_at: datetime | None
    meta: dict[str, Any] | None

    model_config = {"from_attributes": True}


class NotificationListOut(BaseModel):
    items: list[NotificationOut]
    total: int
    unread_count: int


class UnreadCountOut(BaseModel):
    count: int

File: frontend/institution/src/api/notifications.ts
import request, { unwrap } from './request'

export interface AdminNotification {
  id: string
  type: string
  title: string
  content: string
  is_read: boolean
  created_at: string
}

export function listNotifications(): Promise<{ items: AdminNotification[]; total: number; unread_count: number }> {
  return unwrap(request.get('/notifications/?limit=50'))
}
export function markRead(id: string): Promise<AdminNotification> {
  return unwrap<AdminNotification>(request.patch(`/notifications/${id}/read`))
}
export function unreadCount(): Promise<{ count: number }> {
  return unwrap(request.get('/notifications/unread-count'))
}
